ui/src/app/zones/views/ZoneDetails/ZoneDetailsHeader/ZoneDetailsHeader.tsx

```diff
@@ -1,45 +1,84 @@
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 
 import { Button } from "@canonical/react-components";
 import { useSelector, useDispatch } from "react-redux";
+import { useHistory } from "react-router-dom";
+
+import DeleteConfirm from "./DeleteConfirm";
 
 import SectionHeader from "app/base/components/SectionHeader";
 import type { RootState } from "app/store/root/types";
-import { actions } from "app/store/zone";
+import { actions as zoneActions } from "app/store/zone";
 import zoneSelectors from "app/store/zone/selectors";
+import zonesURLs from "app/zones/urls";
 
 type Props = {
   id: number;
 };
 
 const ZoneDetailsHeader = ({ id }: Props): JSX.Element => {
+  const [showConfirm, setShowConfirm] = useState(false);
   const zonesLoaded = useSelector(zoneSelectors.loaded);
   const zone = useSelector((state: RootState) =>
     zoneSelectors.getById(state, Number(id))
   );
   const dispatch = useDispatch();
+  const history = useHistory();
+
+  useEffect(() => {
+    dispatch(zoneActions.fetch());
+  }, [dispatch]);
+
+  const deleteZone = () => {
+    dispatch(zoneActions.delete(id));
+    // we need to cleanup here but don't know how!
+    history.push({ pathname: zonesURLs.index });
+  };
+
```

Comment:
```suggestion
  const [showConfirm, setShowConfirm] = useState(false);
  const zonesLoaded = useSelector(zoneSelectors.loaded);
  const zonesSaved = useSelector(zoneSelectors.saved);
  const zone = useSelector((state: RootState) =>
    zoneSelectors.getById(state, Number(id))
  );
  const dispatch = useDispatch();
  const history = useHistory();

  useEffect(() => {
    dispatch(zoneActions.fetch());
  }, [dispatch]);

  useEffect(() => {
    if(zonesSaved){
      dispatch(zoneActions.cleanup());
       history.push({ pathname: zonesURLs.index });
     }
  }, [dispatch, zonesSaved ]);

  const deleteZone = () => {
    dispatch(zoneActions.delete(id));
  };

```

This is what I tried to suggest yesterday. Hope it helps

Reply:
Thanks! Makes sense